common/src/main/java/tds/common/configuration/RestTemplateConfiguration.java

```diff
@@ -1,14 +1,29 @@
 package tds.common.configuration;
 
-import org.springframework.boot.web.client.RestTemplateBuilder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.client.BufferingClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
 import org.springframework.web.client.RestTemplate;
 
+import java.util.Arrays;
+
+import tds.common.web.interceptors.RestTemplateLoggingInterceptor;
+
 @Configuration
+@Import(JacksonObjectMapperConfiguration.class)
 public class RestTemplateConfiguration {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
     @Bean
-    public RestTemplate restTemplate(RestTemplateBuilder builder) {
-        return builder.build();
+    public RestTemplate restTemplate() {
```

Comment:
I recommend either:
1. using a RestTemplateCustomizer as Todd pointed out OR
2. injecting the RestTemplateBuilder (which comes pre-configured with message converters) and adding an interceptor to the built RestTemplate.

An example from a previous project:
```
@Bean
    @ConditionalOnMissingBean(RestTemplate.class)
    public RestTemplate restTemplate(final RestTemplateBuilder builder,
                                     final ResponseErrorHandler responseErrorHandler,
                                     final ClientHttpRequestFactory clientHttpRequestFactory) {
        final RestTemplate template = builder
                .errorHandler(responseErrorHandler)
                .build();

        // gives us finer grained control over the client behaviour
        template.setRequestFactory(new BufferingClientHttpRequestFactory(clientHttpRequestFactory));

        // return it
        return template;
    }
```